_posts/2020-09-07-introduction-to-system-tests.md

```diff
@@ -0,0 +1,199 @@
+        assertDoesNotThrow(() -> createDelete(modelStr), 
+"Create delete failed after first round-trip -- maybe a problem with a defaulted value?\nApplied string: " + modelStr);
+    }
+```
+
+You can see that the test case basically apply the YAML representation of Kafka and except that will not fail and everything
+is correct. This is applied for every custom resource supported by `Strimzi` for instance `KafkaConnect`, `KafkaMirrorMaker`, 
+`KafkaBridge`, `KafkaTopic`, `KafkaUser` and so on.
+
+After this level is successfully completed we can talk about our system tests. Let's firstly see the informal definition:
+
+>  System testing is a level of testing that validates the complete and fully integrated software product. The purpose 
+of a system test is to evaluate the end-to-end system specifications. Usually, the software is only one element of a 
+larger computer-based system. Ultimately, the software is interfaced with other software/hardware systems. System Testing 
+is actually a series of different tests whose sole purpose is to exercise the full computer-based system. (Guru)
+
+In our case the system tests validates the whole components/features which Strimzi provides. Everything is tested in 
+Kubernetes environment to fully represent production or user environment. Again to clarify I have an example: 
+
+```
+@Test
+void testReceiveSimpleMessage() {
+    KafkaTopicResource.topic(CLUSTER_NAME, TOPIC_NAME).done();
+
+    kafkaBridgeClientJob.consumerStrimziBridge().done();
+
+    // Send messages to Kafka
+    InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+        .withTopicName(TOPIC_NAME)
+        .withNamespaceName(NAMESPACE)
+        .withClusterName(CLUSTER_NAME)
+        .withMessageCount(MESSAGE_COUNT)
+        .withKafkaUsername(USER_NAME)
+        .withUsingPodName(kafkaClientsPodName)
+        .build();
+
+    assertThat(internalKafkaClient.sendMessagesPlain(), is(MESSAGE_COUNT));
+
+    ClientUtils.waitForClientSuccess(consumerName, NAMESPACE, MESSAGE_COUNT);
+}
+
+@BeforeAll
+void createClassResources() throws Exception {
+    deployClusterOperator(NAMESPACE);
+    LOGGER.info("Deploy Kafka and KafkaBridge before tests");
+    // Deploy kafka
+    KafkaResource.kafkaEphemeral(CLUSTER_NAME, 1, 1).done();
+
+    KafkaClientsResource.deployKafkaClients(false, KAFKA_CLIENTS_NAME).done();
+
+    kafkaClientsPodName = kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME).get(0).getMetadata().getName();
+
+    // Deploy http bridge
+    KafkaBridgeResource.kafkaBridge(CLUSTER_NAME, KafkaResources.plainBootstrapAddress(CLUSTER_NAME), 1)
+        .editSpec()
+            .withNewConsumer()
+                .addToConfig(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+            .endConsumer()
+        .endSpec()
+        .done();
+}
+```  
+
+This test case as you can ensuring that we are able to receive messages from `KafkaBridge` component. In the setup phase 
```

Comment:
```suggestion
This test case as you can see ensuring that we are able to receive messages from `KafkaBridge` component. In the setup phase 
```

Reply:
Maybe I'm reading it bad.

Reply:
Good catch! I would say it should be `This test case as you can see ensures that...`

Reply:
That's my czenglish :D but yes, I would fix it that way too